projects/public-search/src/app/patron-profile/patron-profile-loans/patron-profile-loans.component.ts

```diff
@@ -103,6 +110,18 @@ export class PatronProfileLoansComponent implements OnInit, OnDestroy {
   private _loanQuery(page: number): Observable<Record | Error> {
     const patronPid = this._patronProfileMenuService.currentPatron.pid;
     return this._loanApiService
-      .getOnLoan(patronPid, page, this._paginator.getRecordsPerPage());
+      .getOnLoan(patronPid, page, this._paginator.getRecordsPerPage(), undefined, this.sortCriteria);
+  }
+
+  /**
+   * Allow to sort loans list using a sort criteria
+   * @param sortCriteria: the sort criteria to use for sorting the list
+   */
+  selectingSortCriteria(sortCriteria: string) {
+    this.sortCriteria = sortCriteria;
+    this._subscription.unsubscribe();
+    this._initialisePaginatorAndSubscription();
```

Comment:
Use le current paginator and change only le value. Not necessary to reinitialize the paginator.